### PR TITLE
[SPARK-30506][SQL][DOC] Document for generic file source options/configs

### PR DESCRIPTION
**Pull request by xuanyuanking**
All credit to @Ngone51, Closes #27302.

### What changes were proposed in this pull request?
Add a new document page named `Generic File Source Options` for Data Sources menu and added following sub items:

- spark.sql.files.ignoreCorruptFiles
- spark.sql.files.ignoreMissingFiles
- pathGlobFilter
- recursiveFileLookup

And here're snapshots of the generated document:
image
image

### Why are the changes needed?
Better guidance for end-user.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Pass Jenkins.

### CODE REVIEW

**Comment by xuanyuanking** [comment]
cc @Ngone51 @MaxGekk @cloud-fan @gengliangwang

File: docs/sql-data-sources-generic-options.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+These generic options/configurations are effective only when using file-based sources: parquet, orc, avro, json, csv, text.
+
+Please note that the hierarchy of directory used in examples below is:
+
+{% highlight text %}
+
+dir1/
+ ├── dir2/
+ │    └── file2.parquet (schema: <file, string>, content: "file2.parquet")
```
Comment (cloud-fan) [inline docs/sql-data-sources-generic-options.md]: nit: `schema: <file: string>`

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java [hunk 2]
```diff
@@ -106,6 +107,43 @@ public static void main(String[] args) {
...
+    // +-------------+
+    // |file1.parquet|
+    // |file2.parquet|
+    // +-------------+
+    // $example off:ignore_corrupt_files$
+    spark.sql("set spark.sql.files.ignoreMissingFiles=false");
+    // $example on:load_with_path_glob_filter$
+    Dataset<Row> partitionedUsersDF = spark.read().format("orc")
+            .option("pathGlobFilter", "*.orc")
+            .load("examples/src/main/resources/partitioned_users.orc");
```
Comment (cloud-fan) [inline examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: shall we show the result like the previous example?

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java [hunk 3]
```diff
@@ -106,6 +107,43 @@ public static void main(String[] args) {
...
+    // $example off:load_with_path_glob_filter$
+    // $example on:recursive_file_lookup$
+    Dataset<Row>  recursiveLoadedDF = spark.read().format("parquet")
```
Comment (cloud-fan) [inline examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: nit: one space between type definition and `recursiveLoadedDF`

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java (hunk 2 above)
Comment (cloud-fan) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: this is from an existing example. I think we should rewrite it to use the same file set. How about using `*.parquet`?

Comment (cloud-fan) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: and see if we can remove `partitioned_users.orc`

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java [hunk 4]
```diff
@@ -106,6 +107,43 @@ public static void main(String[] args) {
...
+            "examples/src/main/resources/dir1/dir2/");
+    testCorruptDF.show();
+    // +-------------+
+    // |         file|
+    // +-------------+
+    // |file1.parquet|
+    // |file2.parquet|
+    // +-------------+
+    // $example off:ignore_corrupt_files$
+    spark.sql("set spark.sql.files.ignoreMissingFiles=false");
```
Comment (gengliangwang) [inline examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: Why do we need this

Comment (cloud-fan) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: turn off the feature to not affect the following examples that test other features.

File: docs/sql-data-sources-generic-options.md (hunk 1 above)
Comment (xuanyuanking) [reply docs/sql-data-sources-generic-options.md]: Thanks, done in 1de38be.

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java (hunk 2 above)
Comment (xuanyuanking) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: After checking the partitioned_users.orc, it's a demo for hive partitioned table, add the description for `partitioned_users.orc` in the doc. Done in 1de38be.

File: examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java (hunk 4 above)
Comment (gengliangwang) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: I think it is confusing here. The conf `spark.sql.files.ignoreMissingFiles` is disabled by default.

Comment (xuanyuanking) [reply examples/src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java]: I guess setting the config explicitly here is for corresponding with the doc: https://github.com/apache/spark/pull/27415/files#diff-dab20ee598064a54fecef8842b972944R72

**Comment by xuanyuanking** [comment]
@Ngone51 is back, this will be done in #27302.

**Pull request closed without merging**